tool/ood-gen/lib/news.ml

```diff
@@ -1,142 +1,38 @@
-              | Some (Guid_name s) ->
-                Some s
-              | None ->
-                print_endline "No guid";
-                None
-            in
-            try
-              let guid = Option.get guid in
-              let title = item.Ocamlrss.Rss.item_title |> Option.get in
-              let slug = Utils.slugify title in
-              let desc = item.Ocamlrss.Rss.item_desc in
-              let data = item.Ocamlrss.Rss.item_content in
-              let content =
-                match desc, data with
-                | Some _, Some data ->
-                  data
-                | Some desc, None ->
-                  desc
-                | None, Some data ->
-                  data
-                | None, None ->
-                  print_endline "No description or content:encoded";
-                  raise (Invalid_argument "")
-              in
-              let url =
-                item.Ocamlrss.Rss.item_link |> Option.get |> Uri.to_string
-              in
-              let date =
-                item.Ocamlrss.Rss.item_pubdate |> Option.get |> Ptime.to_rfc3339
-              in
-              if List.mem guid source.articles then (
-                let oc =
-                  open_out ("data/news/" ^ source.tag ^ "/" ^ slug ^ ".md")
-                in
-                let Website_meta.{ image; description; _ } =
-                  Website_meta.all url
-                in
-                let metadata =
-                  { title; url; date; preview_image = image; description }
-                in
-                let s = Format.asprintf "%a\n%s\n" pp_meta metadata content in
-                Printf.fprintf oc "%s" s;
-                close_out oc)
-            with
-            | Invalid_argument _ ->
-              Printf.printf
-                "Skipping article %s\n"
-                (Option.value guid ~default:"<no guid>");
-              ())
-          items
-      in
-      ())
-    sources.sources
-
 type t =
   { title : string
-  ; slug : string
-  ; url : string
-  ; description : string option
+  ; description : string
   ; date : string
-  ; preview_image : string option
+  ; slug : string
+  ; tags : string list
   ; body_html : string
   }
 [@@deriving yaml]
 
 let all () =
-  Utils.map_files
-    (fun content ->
+  Utils.map_files_with_names
+    (fun (fname, content) ->
+      let slug = Filename.basename (Filename.remove_extension fname) in
       let metadata, body = Utils.extract_metadata_body content in
       let metadata = Utils.decode_or_raise metadata_of_yaml metadata in
       { title = metadata.title
-      ; slug = Utils.slugify metadata.title
+      ; slug
       ; description = metadata.description
-      ; url = metadata.url
       ; date = metadata.date
-      ; preview_image = metadata.preview_image
-      ; body_html = String.trim body
+      ; tags = metadata.tags
+      ; body_html = Omd.to_html (Omd.of_string (String.trim body))
```

Comment:
Might be worth a `Hilite` call here to get the OCaml code blocks (which I assume would be fairly frequent). Alternatively, it might be better to do this when saving the markdown file since it will replace code blocks with HTML which is still valid markdown? Up to you, we can leave it as is too :))

Reply:
Good idea, will do that!